### PR TITLE
docs: adding smart contract entry to glossary

### PR DESCRIPTION
**Pull request by amessbee**
Adding a glossary entry for smart contracts.

**ref(s):**
[https://papers.agoric.com/smart-contracts-encyclopedia-entry/](https://papers.agoric.com/smart-contracts-encyclopedia-entry/)
[https://github.com/Agoric/documentation/issues/62](https://github.com/Agoric/documentation/issues/62)
[https://github.com/Agoric/agoric-sdk/issues/244](https://github.com/Agoric/agoric-sdk/issues/244)

### CODE REVIEW

**Comment by amessbee** [comment]
@dckc please comment. I think I have added more than what was necessary, but I think the entry itself has to be there. I have deliberately added a line there in order to close #62 :-)

File: main/glossary/index.md [hunk 1]
```diff
@@ -602,6 +602,27 @@ Secure ECMAScript has been renamed to [Hardened JavaScript](#hardened-javascript
 
 An imaginary currency Agoric documentation uses in examples.
 
+## Smart Contract
+
+A smart contract is a self-executing contract with the terms of the agreement directly written into lines of code. These contracts are stored and executed on a blockchain network. Smart contracts automatically enforce and execute the terms of a contract when predefined conditions are met, eliminating the need for intermediaries and reducing the risk of human error or manipulation. Like user accounts on a chain, smart contracts can "own" or hold digital assets themselves.
```
Comment (dckc) [inline main/glossary/index.md:607]: This is not the definition we use at Agoric.

Our definition is:

> A smart contract is a contract-like arrangement expressed in code, where the behavior of the program enforces the terms of the contract.

 - https://papers.agoric.com/smart-contracts-encyclopedia-entry/

File: main/glossary/index.md [hunk 2]
```diff
@@ -602,6 +602,27 @@ Secure ECMAScript has been renamed to [Hardened JavaScript](#hardened-javascript
...
+
+Key Characteristics:
+
+- Automated Execution: Smart contracts automatically execute actions when the conditions specified in the code are satisfied.
+- Transparency: The contract's code is usually transparent and accessible to all participants on the blockchain.
+- Immutability: Once deployed on the blockchain, the contract code cannot be altered, ensuring that the terms remain consistent and tamper-proof.
```
Comment (dckc) [inline main/glossary/index.md:613]: we have governance, so not all our contracts are immutable

**Review by dckc — CHANGES_REQUESTED** [review]

**Comment by amessbee** [comment]
@dckc have a look at updates. Used text from the reference you provided this time.

File: main/glossary/index.md [hunk 3]
```diff
@@ -602,6 +602,27 @@ Secure ECMAScript has been renamed to [Hardened JavaScript](#hardened-javascript
...
+can simply walk away from the exchange with both parties’ assets. 
+
+Smart contracts solve this problem of opportunism during an exchange by providing an 
+[escrow mechanism](#escrow) for the assets from both parties, only dispersing the assets after both
+parties have contributed. There is no need for a human intermediary to perform the escrow; the smart
+contract itself holds the assets. Because the behavior of the smart contract is fully described by
+its code, and that code runs on a decentralized blockchain with high integrity execution, the
+parties can be assured that the code describes exactly what will happen. As such, smart contracts
+hold the promise of being a market-​based solution for the opportunism problem that does not rely on
+State intervention.
```
Comment (dckc) [inline main/glossary/index.md:621]: I'd rather skip the commentary on social problems and State intervention in this context:

```suggestion
A smart contract is a contract-​like arrangement expressed in code, where the behavior of the 
program enforces the terms of the contract.
```

In general, the glossary shouldn't have novel content. The terms should be defined in sections that discuss them. Each glossary entry should be a short gloss with a link to the relevant section.

Here we have a link to `#offer-safety`, which is perhaps good enough. But better would be a link to [Smart Contract Basics](https://docs.agoric.com/guides/zoe/contract-basics.html#smart-contract-basics).

**Review by dckc — CHANGES_REQUESTED** [review]

**Review by dckc — APPROVED** [review]

**Pull request merged**